### ISSUE TITLE
There are many Prometheus Metrics names.

### ISSUE THREAD

**Issue by thinker0** [apache/pulsar#10916]
**BUG REPORT**
There are many, many, many Prometheus Metrics names.

***Describe the bug***
```
^(bookkeeper_client_per_channel_bookie_client|bookkeeper_server_BookieHighPriorityThread|replication_bookkeeper_client_per_channel_bookie_client)_hostnames_port_.*
```

bookies hosts by metrics grow exponentially.

***To Reproduce***

***Expected behavior***
Prometheus OOM (200Gbyte)

***Screenshots***
![image](https://user-images.githubusercontent.com/357785/121538948-2ad12780-ca40-11eb-92e4-4041b5efe785.png)
![image](https://user-images.githubusercontent.com/357785/121539143-5e13b680-ca40-11eb-9883-fdf8874853e5.png)

***Additional context***
https://github.com/apache/bookkeeper/issues/2736

1. Update bookeeper-4.14.1 version

**Comment by codelipenghui** [comment]
The issue had no activity for 30 days, mark with Stale label.

**Comment by Shoothzj** [comment]
have been fixed in 2.8
